aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -23,6 +23,19 @@ void FunctionalTensorWrapper::set_constructor_metadata() {
   storage_ = functional_storage;
   storage_access_should_throw_ = false;
   key_set_ = c10::DispatchKeySet(c10::DispatchKey::Functionalize) | value_.key_set();
+  // What's happening here? In general, functional wrappers *should* copy the keys from their inner tensor.
+  // We need this for the backend use case (LTC/XLA), where any functionalities that the backend uses,
+  // like autograd or autocast, need to run *directly* on the wrapper.
+  // However, all of the keys corresponding to functorch transforms should not be copied over.
+  // Functorch transforms all have their own wrapper tensors (e.g. BatchedTensorImpl) which expect
+  // to participate in the functorch transforms.
```

Comment:
To clarify: the problem is when there exists an invariant that if a Tensor has a Dispatch Key, then it must be have a specific TensorImpl. This occurs a lot in functorch (BatchedTensorImpl <-> FuncTorchBatched key, TensorWrapper <-> FuncTorchGradWrapper key), but is not a functorch-specific problem, right? 

Here are some things that I thought were the case, but I might be wrong about them since it's been a while since I looked into them:
- SparseTensorImpl <-> {Some sparse keys} 
- SparseCsrTensorImpl <-> {the csr sparse keys}
- LTCTensorImpl <-> Lazy

Reply:
Yep that's right - although in practice it isn't a problem for the backend keys, because we're guaranteed to have unwrapped the `FunctionalTensorWrapper` before hitting the backend.

I put used `FuncTorchDynamicLayerFrontMode - FuncTorchDynamicLayerBackMode` because I figured it would be unlikely that we'd add another non-backend, non-functorch key that this logic would apply to, and it lets us avoid hardcoding a new list of keys somewhere.

But I totally forgot that the autograd keys fall in that range. So agreed - I'll create a new keyset corresponding to "tensor impl subclass" keys and use that instead (and at that point I may as well lump in the backend keys, even if they don't directly cause problems in this case).

Reply:
Oh I see, I didn't realize FunctionalTensorWrapper doesn't have any problems with the backend keys that correspond 1:1 with tensor impl subclasses.

The new keyset including all "tensor impl subclass keys" sounds reasonable; if you wanted to just continue on the original train of thought we could create a "FuncTorchKeySet" keyset and put all the dispatch keys that begin with "FuncTorch" into it and use that instead